internals/feature_helpers.py

```diff
@@ -409,24 +408,31 @@ def get_feature_names_by_ids(feature_ids: list[int],
                      for f in cached_features.values()
                      if f is not None and f.get('id')}
 
-  for feature_id in feature_ids:
-    if feature_id not in result_dict:
-      futures_by_id[feature_id] = FeatureEntry.get_by_id_async(feature_id)
+  needed_ids = [fid for fid in feature_ids if fid not in result_dict]
 
-  for future in futures_by_id.values():
-    fe: Optional[FeatureEntry] = future.get_result()
-    if fe and not fe.deleted:
-      feature_id = fe.key.integer_id()
-      result_dict[feature_id] = converters.feature_entry_to_json_tiny(fe)
+  if needed_ids:
+    # Create ndb.Key objects for the batch operation
+    needed_keys = [ndb.Key(FeatureEntry, fid) for fid in needed_ids]
 
-  if update_cache:
+    features_future = FeatureEntry.query(
+        FeatureEntry.key.IN(needed_keys)
```

Comment:
Do we anticipate there ever being more than 30 keys? There's a limit to how many keys you can use. You would need to batch them.

> Use the IN operator to combine up to 30 equality (==) clauses on the same property with a logical OR. An IN query returns entities where the given property matches any of the comparison values.

https://cloud.google.com/datastore/docs/concepts/queries#in